src/ResponseBuilder.cpp

```diff
@@ -0,0 +1,176 @@
+#include "ResponseBuilder.hpp"
+
+/**
+ * @brief Construct a new ResponseBuilder object
+ *
+ * @param configFile Configuration file.
+ * @param fileSystemPolicy File system policy. Can be mocked if needed.
+ */
+ResponseBuilder::ResponseBuilder(const ConfigFile& configFile, const FileSystemPolicy& fileSystemPolicy)
+	: m_configFile(configFile)
+	, m_activeServer(configFile.serverConfigs.begin())
+	, m_fileSystemPolicy(fileSystemPolicy)
+	, m_isFirstTime(true)
+{
+	initMIMETypes();
+}
+
+/**
+ * @brief Set the active server.
+ *
+ * The active server is used for location searching.
+ * @param activeServer Server to be set as active.
+ */
+void ResponseBuilder::setActiveServer(const std::vector<ServerConfig>::const_iterator& activeServer)
+{
+	m_activeServer = activeServer;
+}
+
+/**
+ * @brief Get the built response.
+ *
+ * The stream is returned as c_str() since it is reused.
+ * If str() would be returned, old parts of the stream would be included.
+ * @return std::string Response.
+ */
+std::string ResponseBuilder::getResponse() const { return m_response.str().c_str(); }
+
+/**
+ * @brief Build the response for a given request.
+ *
+ * Reset the stream, to clear possible beforehand built response.
+ * Init the HTTP Response object with data from the request.
+ * If StatusCode is StatusOK: Execute target resource handler to get the target resource.
+ * Else skip this step.
+ * Execute response body handler to get the response body.
+ * Build the response by appending the status line, headers and body.
+ * @param request HTTP request.
+ */
+void ResponseBuilder::buildResponse(const HTTPRequest& request)
+{
+	resetStream();
+
+	HTTPResponse response = initHTTPResponse(request);
```

Comment:
m_response is the reused stringstream. I could have also use a HTTPResponse struct as member. But since I'd planned to construct the ResponseBuilder only once, I would need to clear the struct for every new response. This seems more error prone than just creating a new one.

Because of Return Value Optimisation (RVO) this call should cost nothing, since the struct gets initialised with the return value and not copied.

The initHTTPResponse right now is just a converter from Request to Response structs, so maybe it will have more logic, than just copying stuff from and to.

Reply:
Would advise to rename the member to m_responseStream to make its identity more clear. And to differentiate to the struct.

Reply:
Renamed